Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Eel/ElasticSearchQueryBuilder.php

```diff
@@ -78,69 +81,24 @@ class ElasticSearchQueryBuilder implements QueryBuilderInterface, ProtectedConte
 
-
     /**
      * The ElasticSearch request, as it is being built up.
-     * @var array
+     * @var QueryInterface
      */
-    protected $request = array(
-        // http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-request-query.html
-        'query' => array(
-            // The top-level query we're working on is a *filtered* query, as this allows us to efficiently
-            // apply *global constraints* in the form of *filters* which apply on the whole query.
-            //
-            // NOTE: we do NOT add a search request FILTER to the query currently, because that would mean
-            // that the filters ONLY apply for query results, but NOT for facet calculation (as explained on
-            // http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-request-filter.html)
-            //
-            // Reference: http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/query-dsl-filtered-query.html
-            'filtered' => array(
-                'query' => array(
-                    'bool' => array(
-                        'must' => array(
-                            array(
-                                'match_all' => array()
-                            )
-                        )
-                    )
-
-                ),
-                'filter' => array(
-                    // http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/query-dsl-bool-filter.html
-                    'bool' => array(
-                        'must' => array(),
-                        'should' => array(),
-                        'must_not' => array(
-                            // Filter out all hidden elements
-                            array(
-                                'term' => array('_hidden' => true)
-                            ),
-                            // if now < hiddenBeforeDateTime: HIDE
-                            // -> hiddenBeforeDateTime > now
-                            array(
-                                'range' => array('_hiddenBeforeDateTime' => array(
-                                    'gt' => 'now'
-                                ))
-                            ),
-                            array(
-                                'range' => array('_hiddenAfterDateTime' => array(
-                                    'lt' => 'now'
-                                ))
-                            ),
-                        ),
-                    )
-                )
-            )
-        ),
-        'fields' => array('__path')
-    );
-
+    protected $request;
 
     /**
      * @var array
      */
     protected $result = array();
 
+    /**
+     * Initialize Object
+     */
+    public function initializeObject()
+    {
+        $this->request = $this->objectManager->get('Flowpack\ElasticSearch\ContentRepositoryAdaptor\Domain\Model\QueryInterface', $this);
```

Comment:
There are multiple implementations of that interface, how should the ObjectManager know which one to pick?